src/sensors/lorreconstruction.rs

```diff
@@ -19,6 +19,31 @@ use crate::config::mlem::Bounds;
 
 type DetectionEff = f32;
 
+// Define a uom version of the speed of light and some helpers.
+// This should probably go in its own file/crate but since it's
+// just the one for now, here.
+// Debatable whether we need the physical_constants crate since
+// uom seems to force f32 which loses some accuracy anyway and
+// it's unlikely that any updates will significantly affect us.
```

Comment:
> Note: we already have the speed of light [here](https://github.com/jacg/petalo/blob/master/src/exports.rs#L44).

Oops, missed that one. It made me read more about `uom` etc so not a complete loss. Also, the value doesn't seem to be correct: 

`use crate::C;
    #[test]
    fn test_stuff() {
        println!("C check: {:?}", C);
        assert_eq!(1, 2)
    }`
gives me `C check: 299792450.0 mm^1 ps^-1`. I tried using the `in_base_unit!` macro in the snippet I wrote and got the same result. I assumed that this was because we define the `f32` units [here](https://github.com/jacg/petalo/blob/e94173c55162f4259a65ee94b47f7fcba61213d8/geometry/src/units.rs#L36) with `millimeter` and `picosecond` as the `Length` and `Time` units but I have to admit that that part of the documentation was not an easy read.

Reply:
Hmm, there's something fishy going on. I wonder how the reconstruction can be giving us sensible results, if the speed of light is out by a factor of 10^9.

Some background:

+ Our data originates in Geant4 which (implicitly) uses picoseconds and millimeters as the units of time and length. Consequently our original code inherited this implicit meaning of `f32`s used to represent times and lengths. 
+ `uom`, by default, uses `SI` units for its quantities. So the *base unit* of `Length` is `m` and the *base unit* of `Time` is `s`. This means that the `f32` `1.0` stored inside a `Length` means 1 metre.
+ When `uom`-izing our code, initially I used `uom`'s standard units, and got all sorts of annoying discrepancies WRT the pre-`uom` version, because 1 metre was no longer represented by the float `1.0` but by `1000.0`, so all lengths being read in from HDF5 had to be multiplied by 1000 (along with a corresponding adjustment for time). 1000 (powers of 10, in general) is not representable in binary, so this led to tiny (but noticeable in various tests) differences WRT the original version.
+ To get around this, I created our own `uom` set of units`:

   https://github.com/jacg/petalo/blob/e94173c55162f4259a65ee94b47f7fcba61213d8/geometry/src/units.rs#L13-L21

   with `mm` and `ps` as the base units of length and time, respectively.
+ Using these units, no numerical conversion needs to take place when reading HDF5, and the `uom` version gave *exactly* the same results as the pre-`uom` version.
+ Consequently, the base unit of speed, in our units, is `mm/ps`.
+ The trouble with the `in_base_unit!` macro is that the base units are implicit (so it forgoes most of the safety we want from `uom`). I'd prefer not to use the macro, but, at present, there doesn't seem to be a better way of creating a Rust `const` of `uom` type. :-(

Reply:
Very strange. And if you run changing `C` to the expected value?

Reply:
Well, yes, `in_base_unit!` was doing exactly what it was told, which was to set the speed of light to 10^9 times its real value. Or, rather to its value in `m/s` even though I'm fully aware that the base units are `mm/ps`. I guess that I was expecting some tests to catch this problem, and they hadn't, and so it slipped through the net.

Anyway, I've tried to fix this in #24.

Reply:
> And if you run changing C to the expected value?

All the tests still pass.